### ISSUE TITLE
Confused by Size of Confidence Loss

### ISSUE THREAD

**Issue by Paandaman** [amdegroot/ssd.pytorch#410]
In the multibox loss, the Localization loss is calculated as follows:

`loss_l = F.smooth_l1_loss(loc_p, loc_t, size_average=False)`

I need to save the loss corresponding to each sample in each batch, so I'm running:

`loss_l = F.smooth_l1_loss(loc_p, loc_t, size_average=False, reduce=False)`

instead which outputs the loss in size (num_priors, 4). So, number of prior boxes and the error for the x, y, w and h values.
When I try to do the same for the Confidence Ioss, that is:

`loss_c = F.cross_entropy(conf_p, targets_weighted, size_average=False, reduce=False) `

I expected to get a confidence loss for each box, that is with size (num_priors), instead I get the size (num_priors, 4) for this too. Can someone explain what these 4 loss terms comes from?  I thought it was supposed to be the cross entropy loss from the class prediction in that box, but that does not seem to be the case! 

Many thanks in advance!
